subsys/shell/shell.c

```diff
@@ -610,7 +610,7 @@ static int execute(const struct shell *shell)
 	}
 
 	/* Parent present means we are in select mode. */
-	if (parent != NULL) {
+	if (IS_ENABLED(SHELL_CMDS_SELECT) && (parent != NULL)) {
```

Comment:
Should be `CONFIG_SHELL_CMDS_SELECT` instead?

Reply:
Additionally what do you think about introducing:
```
static inline const struct shell_static_entry *
shell_selected_cmd(const struct shell *shell)
{
	if (IS_ENABLED(CONFIG_SHELL_CMDS_SELECT)) {
		return shell->ctx->selected_cmd;
	}

	return NULL;
}
```
so that you can then convert
```
	const struct shell_static_entry *parent = shell->ctx->selected_cmd;
```
to
```
	const struct shell_static_entry *parent = shell_selected_cmd(shell);
```
and leave above `if (parent != NULL) {` untouched?

Seems like using `shell_selected_cmd(shell)`  instead of `shell->ctx->selected_cmd` in other places can save another 16B. Not much, but at least some additional `IS_ENABLED(CONFIG_SHELL_CMDS_SELECT)` can be dropped in code. Look [here](https://github.com/mniestroj/zephyr/commit/05719d8cf12ac9c7e85a8810186f7101a65fad01).

Reply:
good point! I will update it.